rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/IdentifiedArchives.java

```diff
@@ -0,0 +1,83 @@
+package org.jboss.windup.rules.apps.java.archives.identify;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.commons.lang.StringUtils;
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+
+/**
+ * Used to identify JAR files. Maps SHA1 hashes to a {@link Coordinate}.
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ * @author <a href="mailto:lincolnbaxter@gmail.com">Lincoln Baxter, III</a>
+ */
+public class IdentifiedArchives
+{
+    private static final Logger log = Logger.getLogger(IdentifiedArchives.class.getName());
+    private static final Map<String, String> map = new TreeMap<>();
+
+    public static Coordinate getCoordinateFromSHA1(String sha1Hash)
+    {
+        if (sha1Hash == null)
+            return null;
+
+        String coordinate = map.get(sha1Hash);
+
+        if (coordinate == null)
+            return null;
+
+        return CoordinateBuilder.create(coordinate);
+    }
+
+    public static void addMapping(String sha1, String coordinate)
+    {
+        map.put(sha1, coordinate);
+    }
+
+    public static void addMappingsFrom(File file)
+    {
+        try
+        {
+            addMappingsFrom(new FileInputStream(file));
+        }
+        catch (FileNotFoundException ex)
+        {
+            log.log(Level.SEVERE, "Failed loading SHA1 to CoordinateUtils mapping from " + file.toString() + ": " + ex.getMessage(), ex);
```

Comment:
I think this could be thrown instead of just logged. I guess it doesn't matter too much either way, though.